src/reanimated2/animation/sequence.ts

```diff
@@ -1,29 +1,77 @@
-import { defineAnimation } from './util';
+import { defineAnimation, getReduceMotionForAnimation } from './util';
 import type { NextAnimation, SequenceAnimation } from './commonTypes';
 import type {
   Animation,
   AnimatableValue,
   AnimationObject,
+  ReduceMotion,
   Timestamp,
 } from '../commonTypes';
 
-// TODO TYPESCRIPT This is a temporary type to get rid of .d.ts file.
-type withSequenceType = <T extends AnimatableValue>(...animations: T[]) => T;
+export function withSequence<T extends AnimatableValue>(
+  _reduceMotion: ReduceMotion,
+  ...animations: T[]
+): T;
 
-export const withSequence = function (
+export function withSequence<T extends AnimatableValue>(...animations: T[]): T;
+
+export function withSequence(
+  _reduceMotionOrFirstAnimation?: ReduceMotion | NextAnimation<AnimationObject>,
   ..._animations: NextAnimation<AnimationObject>[]
 ): Animation<SequenceAnimation> {
   'worklet';
+  let reduceMotion: ReduceMotion | undefined;
+
+  // the first argument is either a config or an animation
+  // this is done to allow the reduce motion config prop to be optional
+  if (_reduceMotionOrFirstAnimation) {
+    if (typeof _reduceMotionOrFirstAnimation !== 'string') {
+      _animations.unshift(
+        _reduceMotionOrFirstAnimation as NextAnimation<AnimationObject>
+      );
+    } else {
+      reduceMotion = _reduceMotionOrFirstAnimation as ReduceMotion;
+    }
+  }
+
+  if (_animations.length === 0) {
+    console.warn('No animation was provided for the sequence');
```

Comment:
Do you think it makes sense to include library tag in the warning message?
```suggestion
    console.warn('[Reanimated] No animation was provided for the sequence');
```